### DISCUSSION TITLE
Save progress/environment for later (Python + VS Code)

### DISCUSSION THREAD

**Discussion by kvten** [quarto-dev/quarto-cli#3773] [Q&A]
Is it possible to save imported data, variables...  similar to `.Rdata` or console output similar to `Jupyter notebooks` for this setup: ***Quarto, VS Code, Python***?

**Comment by mcanouil** [comment]
I believe you are talking about cache.
See <https://quarto.org/docs/computations/caching.html>.

**Reply by kvten** [reply]
I'm not able to stop and pick up where I left off without having to re-run each code chunk.

### For example:

![Screenshot](https://user-images.githubusercontent.com/103538771/209875003-1833c149-e86c-4fd6-a82c-ffeeee07cf4a.png)

`a` is not saved when I re-open the `.qmd` file.

**Reply by mcanouil** [reply]
I see what you want to do.
That's not how the cache is supposed to work thus is not suited to what you want to achieve, see #3697 which is related.

Since Jupiter notebooks format can be used directly in Quarto, why don't you use it? <https://quarto.org/docs/tools/jupyter-lab.html>

In my opinion, it's not a good practice/idea to use literate programming as a development environment.
